src/incubator/Dialog/useDialogCloseButton.tsx

```diff
@@ -0,0 +1,87 @@
+import React, {PropsWithChildren, useMemo} from 'react';
+import {BorderRadiuses, Colors} from '../../style';
+import {DialogCloseButtonProps} from './types';
+import {StyleSheet} from 'react-native';
+import Assets from '../../assets';
+import DialogHeader from './DialogHeader';
+import Icon from '../../components/icon';
+import Text from '../../components/text';
+import TouchableOpacity from '../../components/touchableOpacity';
+import View from '../../components/view';
+
+interface InternalDialogCloseButtonProps extends PropsWithChildren<DialogCloseButtonProps> {
+  close: () => void;
+}
+
+const useDialogCloseButton = (props: InternalDialogCloseButtonProps) => {
```

Comment:
Please rename it to useDialogContent as you return the whole content and not just the Button